Windows tests can fail during uninstall with "The service cannot accept control messages at this time" causing cascading test failures

## Flaky Test

* **Test Name:** TestStandaloneUpgradeRollbackOnRestarts
* **Link:** https://github.com/elastic/elastic-agent/blob/99b14c8acc63e65c710dca19e2115200209ff314/testing/integration/upgrade_rollback_test.go#L120-L123
* **Branch:** main
* **Artifact Link:** https://buildkite.com/elastic/elastic-agent/builds/3822#018b0428-cb9f-479d-9265-ddc110a25d1a
* **Notes:** It appears something went wrong during this test when it was attempting to restart the agent to simulate a failure, which then lead to every following test also failing.

### Stack Trace

The stack trace of what I think is the first failure is:

```
 fixture.go:365: >> running agent with: [C:\Program Files\Elastic\Agent\elastic-agent.exe upgrade 8.10.1 --source-uri file://C:\Users\windows\agent\.agent-testing\artifact --skip-verify]
    upgrader.go:234: waiting for upgrade watcher to start
    upgrader.go:239: upgrade watcher started
    upgrade_rollback_test.go:171: Restarting Agent via service to simulate crashing
    upgrade_rollback_test.go:171: Restarting Agent via service to simulate crashing
    upgrade_rollback_test.go:171: Restarting Agent via service to simulate crashing
    upgrade_rollback_test.go:176: Got an allowed error on Windows: failed to restart service (Elastic Agent): The service has not been started.
    upgrade_rollback_test.go:171: Restarting Agent via service to simulate crashing
    fixture.go:365: >> running agent with: [C:\Program Files\Elastic\Agent\elastic-agent.exe status --output json]
    fixture.go:365: >> running agent with: [C:\Program Files\Elastic\Agent\elastic-agent.exe status --output json]
    fixture_install.go:119: [test TestStandaloneUpgradeRollbackOnRestarts] Inside fixture cleanup function
    fixture.go:365: >> running agent with: [C:\Program Files\Elastic\Agent\elastic-agent.exe uninstall --force]
    fixture_install.go:169: 
        	Error Trace:	C:/Users/windows/agent/pkg/testing/fixture_install.go:169
        	            				C:/Program Files/Go/src/testing/testing.go:1150
        	            				C:/Program Files/Go/src/testing/testing.go:1328
        	            				C:/Program Files/Go/src/testing/testing.go:1570
        	Error:      	Received unexpected error:
        	            	error running uninstall command: exit status 1
        	Test:       	TestStandaloneUpgradeRollbackOnRestarts
        	Messages:   	uninstalling agent failed. Output: "Stopping service................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................."
--- FAIL: TestStandaloneUpgradeRollbackOnRestarts (478.51s)
```

This then leads to a series of errors installing the agent in the following tests. There are two unique errors that occur, the first is the service not being able to accept control messages:

```
upgrade_standalone_retry_test.go:94: 
        	Error Trace:	C:/Users/windows/agent/testing/integration/upgrade_standalone_retry_test.go:94
        	Error:      	Received unexpected error:
        	            	failed to install start agent (err: error running agent install command: exit status 1) [output: Installing in non-interactive mode.
        	            	Uninstalling current Elastic Agent...
        	            	   Stopping service... FAILED
        	            	   FAILED
        	            	Error: failed to uninstall Agent at (C:\Program Files\Elastic): failed to stop service (Elastic Agent): The service cannot accept control messages at this time.
        	            	For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.10/fleet-troubleshooting.html
        	            	]
        	Test:       	TestStandaloneUpgradeRetryDownload
    upgrade_standalone_retry_test.go:95: 
        	Error Trace:	C:/Users/windows/agent/testing/integration/upgrade_standalone_retry_test.go:95
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 0
        	Test:       	TestStandaloneUpgradeRetryDownload
        	Messages:   	retry request didn't occur
--- FAIL: TestStandaloneUpgradeRetryDownload (29.60
```

The second is related to a failed attempt to acquire a lock file:

```
=== RUN   TestInstallWithoutBasePath
    fetcher.go:90: Using existing artifact elastic-agent-8.10.2-SNAPSHOT-windows-x86_64.zip
    fixture.go:221: Extracting artifact elastic-agent-8.10.2-SNAPSHOT-windows-x86_64.zip to C:\Users\windows\AppData\Local\Temp\TestInstallWithoutBasePath1341857470\001
    fixture.go:234: Completed extraction of artifact elastic-agent-8.10.2-SNAPSHOT-windows-x86_64.zip to C:\Users\windows\AppData\Local\Temp\TestInstallWithoutBasePath1341857470\001
    fixture.go:558: Components were not modified from the fetched artifact
    install_test.go:57: 
        	Error Trace:	C:/Users/windows/agent/testing/integration/install_test.go:57
        	Error:      	Received unexpected error:
        	            	remove C:\Program Files\Elastic\Agent\data\agent.lock: The process cannot access the file because it is being used by another process.
        	Test:       	TestInstallWithoutBasePath
        	Messages:   	failed to remove %q. The test requires this path not to exist.
--- FAIL: TestInstallWithoutBasePath (16.68s)
```

--- a/NOTICE.txt
+++ b/NOTICE.txt
@@ -1367,11 +1367,11 @@ SOFTWARE
 
 --------------------------------------------------------------------------------
 Dependency : github.com/elastic/elastic-agent-libs
-Version: v0.5.0
+Version: v0.6.0
 Licence type (autodetected): Apache-2.0
 --------------------------------------------------------------------------------
 
-Contents of probable licence file $GOMODCACHE/github.com/elastic/elastic-agent-libs@v0.5.0/LICENSE:
+Contents of probable licence file $GOMODCACHE/github.com/elastic/elastic-agent-libs@v0.6.0/LICENSE:
 
                                  Apache License
                            Version 2.0, January 2004

--- a/changelog/fragments/1697662109-upgrade-elastic-agent-libs-to-v0.6.0.yaml
+++ b/changelog/fragments/1697662109-upgrade-elastic-agent-libs-to-v0.6.0.yaml
@@ -1,0 +1,32 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: bug-fix
+
+# Change summary; a 80ish characters long description of the change.
+summary: upgrade elastic-agent-libs to v0.6.0
+
+# Long description; in case the summary is not enough to describe the change
+# this field accommodate a description without length limits.
+# NOTE: This field will be rendered only for breaking-change and known-issue kinds at the moment.
+description: 'allows elastic-agent running as a windows service to receive more than one change request.'
+
+# Affected component; a word indicating the component this changeset affects.
+component: agent
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+#pr: https://github.com/owner/repo/1234
+
+# Issue URL; optional; the GitHub issue related to this changeset (either closes or is part of).
+# If not present is automatically filled by the tooling with the issue linked to the PR number.
+#issue: https://github.com/owner/repo/1234

--- a/go.mod
+++ b/go.mod
@@ -15,7 +15,7 @@ require (
 	github.com/elastic/e2e-testing v1.1.0
 	github.com/elastic/elastic-agent-autodiscover v0.6.4
 	github.com/elastic/elastic-agent-client/v7 v7.4.0
-	github.com/elastic/elastic-agent-libs v0.5.0
+	github.com/elastic/elastic-agent-libs v0.6.0
 	github.com/elastic/elastic-agent-system-metrics v0.7.0
 	github.com/elastic/elastic-transport-go/v8 v8.3.0
 	github.com/elastic/go-elasticsearch/v8 v8.10.0

